### PR TITLE
Contribute page

### PR DESCRIPTION
**Pull request by dannify**
New contribution guide with all the info

### CODE REVIEW

File: packages/dev/docs/pages/contribute.mdx [hunk 1]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
 ---
 
 # Contribute
+
+Thanks for choosing to contribute! We look forward to improving web applications together. Here you will find information on how propose bug fixes, suggest improvements and how to build react-spectrum yourself.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:20]:
```suggestion
Thanks for choosing to contribute! We look forward to improving web applications together. Here you will find information on how propose bug fixes, suggest improvements, and develop locally.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 2]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+## Better together
+We believe in sharing our process, results and journey with you all. The core team and all external contributors follow the same process in order maintain a high quality codebase.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:23]:
```suggestion
We believe that the best way to build a better web is together as a community. The React Spectrum project aims to make it easier to build design systems and component libraries with high quality interactions and accessibility for all. The core team and all external contributors follow the same process in order maintain a high quality codebase.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 3]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+## Code of conduct
+We adhere to the Adobe [code of conduct](https://github.com/adobe-private/react-spectrum-v3/blob/CODE_OF_CONDUCT.md) and by participating, you are expected to uphold this code. Please report unacceptable behavior to Grp-opensourceoffice@adobe.com.
+
+## Reporting issues
+### Bugs
+We use [GitHub issues](https://github.com/adobe-private/react-spectrum-v3/issues) to track work and log bugs. Please check existing issues before filing anything new. We do our best to respond to issues within a few days. If you would like to contribute a fix, please let us know on the issue.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:30]:
```suggestion
We use [GitHub issues](https://github.com/adobe-private/react-spectrum-v3/issues) to track work and log bugs. Please check existing issues before filing anything new. We do our best to respond to issues within a few days. If you would like to contribute a fix, please let us know by leaving a comment on the issue.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 4]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+The best way to reduce back and forth on a bug is to add code for a storybook example exhibiting the issue along with steps to reproduce it. If you would like to work on a bugfix yourself, make sure an issue exists first.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:32]: I think we don't need it to be storybook specifically? Something like a code sandbox link would also work. Just need to be a minimal reproduction.

```suggestion
The best way to reduce back and forth on a bug is provide a small code example exhibiting the issue along with steps to reproduce it. If you would like to work on a bugfix yourself, make sure an issue exists first.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 5]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+### Feature requests
+Our components can always be improved upon. If you have a feature request, you can use our Feature Request issue template in issues. For larger scopes of work, it is a good idea to put forth a Request For Comments (RFC) first to gather feedback from the team. A [template](https://github.com/adobe-private/react-spectrum-v3/blob/master/rfcs/template.md) is available to build an RFC. Make a PR to add your RFC to the [rfcs folder](https://github.com/adobe-private/react-spectrum-v3/tree/master/rfcs) to give the team and the community a chance to discuss the proposal.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:35]:
```suggestion
Our components can always be improved upon. If you have a feature request, you can use our Feature Request issue template. For larger scopes of work, it is a good idea to open a Request For Comments (RFC) first to gather feedback from the team. Please follow our RFC [template](https://github.com/adobe-private/react-spectrum-v3/blob/master/rfcs/template.md). Make a PR to add your RFC to the [rfcs folder](https://github.com/adobe-private/react-spectrum-v3/tree/master/rfcs) to give the team and the community a chance to discuss the proposal.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 6]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+### Security issues
+Security issues shouldn't be reported on this issue tracker. Instead, [file an issue](https://helpx.adobe.com/security/alertus.html) to our security experts.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:38]:
```suggestion
Security issues shouldn't be reported on this issue tracker. Instead, please follow the directions [here](https://helpx.adobe.com/security/alertus.html) to contact our security team.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 7]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+## Code of conduct
+We adhere to the Adobe [code of conduct](https://github.com/adobe-private/react-spectrum-v3/blob/CODE_OF_CONDUCT.md) and by participating, you are expected to uphold this code. Please report unacceptable behavior to Grp-opensourceoffice@adobe.com.
+
+## Reporting issues
+### Bugs
+We use [GitHub issues](https://github.com/adobe-private/react-spectrum-v3/issues) to track work and log bugs. Please check existing issues before filing anything new. We do our best to respond to issues within a few days. If you would like to contribute a fix, please let us know on the issue.
+
+The best way to reduce back and forth on a bug is to add code for a storybook example exhibiting the issue along with steps to reproduce it. If you would like to work on a bugfix yourself, make sure an issue exists first.
+
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:33]: Maybe add something asking that people follow the issue template like you have for PRs?

File: packages/dev/docs/pages/contribute.mdx [hunk 8]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+yarn run
+```
+
+Or run the documentation and browse to [http://localhost:1234/](http://localhost:1234/) with:
+```bash
+yarn start:docs
+```
+
+### Tests
+We use [jest](https://jestjs.io/) for unit tests and [react-testing-library](https://testing-library.com/docs/react-testing-library/intro) for rendering and writing assertions. Please make sure you include tests with your pull requests. Our CI/CD will run the tests on PRs as well as lint code and TS. You can see on each PR whether you have passed all our checks. We split the tests into 2 groups.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:96]:
```suggestion
We use [jest](https://jestjs.io/) for unit tests and [react-testing-library](https://testing-library.com/docs/react-testing-library/intro) for rendering and writing assertions. Please make sure you include tests with your pull requests. Our CI will run the tests on PRs as well as the linter and type checker. You can see on each PR whether you have passed all our checks. We split the tests into 2 groups.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 9]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+*Visual tests*
+- A Storybook story should be written for any visual breakage of a component.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:99]:
```suggestion
- A Storybook story should be written for each visual state that a component can be in (based on props).
```

File: packages/dev/docs/pages/contribute.mdx [hunk 10]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+```
+
+### TypeScript
+The code for React Spectrum is written in [TypeScript](https://www.typescriptlang.org/). The type checker will usually run in your editor, but also runs when you run
+```bash
+yarn lint
+```
+
+### Storybook
+We use [Storybooks](https://storybooks.js.org) for local development. You can use the command
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:130]:
```suggestion
We use [Storybook](https://storybooks.js.org) for local development. Run the following command to start it:
```

File: packages/dev/docs/pages/contribute.mdx [hunk 11]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+```bash
+yarn start
+```
+followed by opening [http://localhost:9003](http://localhost:9003) in your browser to play around with the components and test your changes.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:134]:
```suggestion
Then, open [http://localhost:9003](http://localhost:9003) in your browser to play around with the components and test your changes.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 12]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+### Documentation
+Our documentation should always remain up to date. When making changes to components, make sure the appropriate documentation has been updated to reflect those changes. Documentation for each component can be found in the docs folder within a component's package. Other documentation pages can be found in the [packages/dev/docs](https://github.com/adobe-private/react-spectrum-v3/tree/master/packages/dev/docs) folder in the codebase.
+
+Documentaton can be run locally by using
+```bash
+yarn start:docs
+```
+followed by opening [http://localhost:1234](http://localhost:1234) in your browser.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:143]:
```suggestion
Then, open [http://localhost:1234](http://localhost:1234) in your browser.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 13]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+
+## Contributor to committer
+
+We love contributions from our community! If you'd like to go a step beyond contributor and become a committer with full write access and a say in the project, you must be invited to the project. The existing committers employ an internal nomination process that must reach lazy consensus (silence is approval) before invitations are issued. If you feel you are qualified and want to get more deeply involved, feel free to reach out to existing committers to have a conversation.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:147]:
```suggestion
We love contributions from our community! If you'd like to go a step beyond contributor and become a committer with full write access and a say in the project, you must be invited. The existing committers employ an internal nomination process that must reach lazy consensus (silence is approval) before invitations are issued. If you feel you are qualified and want to get more deeply involved, feel free to reach out to existing committers to have a conversation.
```

File: packages/dev/docs/pages/contribute.mdx [hunk 14]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+- Update missing or fix existing documentation
+- Review and test open pull requests
+
+## Developing
+When you are ready to start developing you can clone the repo and start storybook.
+Make sure you have the following requirements installed: [node](https://nodejs.org/) (v11.0.0+) and [yarn](https://yarnpkg.com/en/) (v1.22.0+)
+
+Fork the repo first using [this guide](https://help.github.com/articles/fork-a-repo), then clone it locally.
+```
+git clone https://github.com/YOUR-USERNAME/react-spectrum-v3
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:80]:
```suggestion
git clone https://github.com/YOUR-USERNAME/react-spectrum
```

File: packages/dev/docs/pages/contribute.mdx [hunk 15]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
+cd react-spectrum-v3
```
Comment (dannify) [inline packages/dev/docs/pages/contribute.mdx:81]:
```suggestion
cd react-spectrum
```

File: packages/dev/docs/pages/contribute.mdx [hunk 16]
```diff
@@ -16,3 +16,132 @@ keywords: [react spectrum contributions, bug reports, feature requests]
...
 
 # Contribute
+
+Thanks for choosing to contribute! We look forward to improving web applications together. Here you will find information on how propose bug fixes, suggest improvements, and develop locally.
+
+## Better together
+We believe that the best way to build a better web is together as a community. The React Spectrum project aims to make it easier to build design systems and component libraries with high quality interactions and accessibility for all. The core team and all external contributors follow the same process in order maintain a high quality codebase.
+
+## Code of conduct
+We adhere to the Adobe [code of conduct](https://github.com/adobe-private/react-spectrum-v3/blob/CODE_OF_CONDUCT.md) and by participating, you are expected to uphold this code. Please report unacceptable behavior to Grp-opensourceoffice@adobe.com.
```
Comment (devongovett) [inline packages/dev/docs/pages/contribute.mdx:26]: Just noticed the link 404s

**Pull request merged**